src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.tt

```diff
@@ -116,96 +116,110 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
 			WriteLine(string.Format("        public override IEnumerable<string> ApplicableParents() {{ return new[] {{ \"{0}\" }}; }}", type));
 
 			List<XmlElement> children = GetChildren(xmlFile, type);
-			List<XmlElement> filters = GetNodeFilters(xmlFile, type);
 			List<XmlElement> smoProperties = GetNodeSmoProperties(xmlFile, type);
 
-			if (filters.Count > 0)
+			// Load and parse Filters node
+			// A <Filters> node is comprised of <Or> and <Filter> nodes
+			//   - A <Filter> node defines the properties to construct a NodePropertyFilter object
```

Comment:
It seems like we should probably rename `Filter` to `PropertyFilter` to make it clear what type of filter it is since we have multiple types right now.

Reply:
On second thought, that can be a follow-up item. Will help avoid making this PR even more messy than it already is.